reconcile/saas_auto_promotions_manager/subscriber.py

```diff
@@ -142,22 +144,42 @@ def _compute_desired_ref(self) -> None:
                     break
                 publisher_refs.add(publisher.commit_sha)
 
+        # By default we keep current state
+        self.desired_ref = self.ref
+
         if len(publisher_refs) > 1:
             logging.info(
                 "Publishers for subscriber at path %s have mismatching refs: %s",
                 self.target_file_path,
                 publisher_refs,
             )
-        if (
-            len(publisher_refs) != 1
-            or any_bad_deployment
-            or not self._passed_accumulated_soak_days()
-        ):
-            # We keep current state
-            self.desired_ref = self.ref
-        else:
-            # We have a common single publisher ref w/o any deployment issues
-            self.desired_ref = next(iter(publisher_refs))
+            return
+
+        if not self._passed_accumulated_soak_days():
+            logging.info(
```

Comment:
debug

Reply:
Good point. Added debug level for soak days